### PR TITLE
FEAT(client): Add MP3 Recording

### PR DESCRIPTION
**Pull request by Chinry**
implements #3079

### Checks

- [ ] My commits follow the [commit guidelines](https://github.com/mumble-voip/mumble/blob/master/COMMIT_GUIDELINES.md)

### CODE REVIEW

**Comment by Hartmnt** [comment]
@Chinry You can simply amend your changes and force push the branch :D There is no need to create new pull request for your changes. It helps to keep track of the discussion, you know.

**Comment by Chinry** [comment]
@Hartmnt, when I reset my changes it closed the request. Would it automatically reopen it once I pushed?

**Comment by Hartmnt** [comment]
Well, I do not really know what exactly you are doing, but it should be sufficient to do the following:

1) Change files
2) ``git add .`` (You may optionally replace ``.`` with any other file or folder)
3) ``git commit --amend`` (Here you can also change the commit message)
4) ``git push origin record-mp3`` (where ``record-mp3`` is the branch name linked in the pull request)

That should push your changes to the existing pull request without closing it.

**Comment by Hartmnt** [comment]
Other than that it looks good. The translation commit is missing, of course. I will explain to you how to add the translation commit once I have had the time to test whether this new feature actually works ;)

**Comment by Chinry** [comment]
I appreciate the help with git. I made those 2 minor changes.

**Comment by Chinry** [comment]
@Hartmnt, how do I make a translation commit?

**Comment by Hartmnt** [comment]
> @Hartmnt, how do I make a translation commit?

You run the ``scripts/updateTranslations.py`` script from the repo root. It automatically creates the commit.

However, for one I have not had the time to try your changes myself yet. And also your formatting is still failing (see the PR-Checks job log) and it would be (slightly) harder to update your commit when you have added the translation commit (``git commit --amend`` would no longer work and you would have to fiddle with rebase)

**Comment by Chinry** [comment]
Not sure while it is failing after I ran _clang-format_ on the file. It seems to have deleted some excess spaces that cause it to fail around line 81.

**Comment by Hartmnt** [comment]
> Not sure while it is failing after I ran _clang-format_ on the file. It seems to have deleted some excess spaces that cause it to fail around line 81.

Mhh, interesting. I assume you used clang-format in version 10? Anyway, it should be sufficient to revert to the line with the spaces intact. After all your main code changes did not even touch line 81.

**Comment by Chinry** [comment]
Fixed the formatting issue. Let me know when you get around to taking a look at the changes.

File: src/mumble/VoiceRecorder.cpp [hunk 1]
```diff
@@ -277,6 +289,16 @@ bool VoiceRecorder::ensureFileIsOpenedFor(SF_INFO &soundFileInfo, boost::shared_
 		return false;
 	}
 
+#ifdef USE_SNDFILE_MP3
+	// Enable constant 320kbps for MP3 formats
+	if (m_config.recordingFormat == VoiceRecorderFormat::MP3) {
+		int bitrate        = SF_BITRATE_MODE_CONSTANT;
+		double compression = 0.0;
+		sf_command(ri->soundFile, SFC_SET_BITRATE_MODE, &bitrate, sizeof(int));
+		sf_command(ri->soundFile, SFC_SET_COMPRESSION_LEVEL, &compression, sizeof(double));
```
Comment (Krzmbrzl) [inline src/mumble/VoiceRecorder.cpp:298]: I'm wondering whether these changes might have an effect on subsequent recordings where the chosen format is something other than MP3? Have you tested that?

Comment (Chinry) [reply src/mumble/VoiceRecorder.cpp:298]: Should I just get rid of it then? I think you're right about it having effects on future recordings.

Comment (Krzmbrzl) [reply src/mumble/VoiceRecorder.cpp:298]: I don't know. You probably added them for a reason, no? If they don't do anything, then remove them, but if they are important/useful for MP3 recordings, they should be kept and some mechanism of restoring the previous settings when using other recording formats should be implemented :thinking:

**Review by Krzmbrzl — APPROVED** [review]
I assume you have actually verified that recording to MP3 works with this code?
If that's true, then this LGTM. I'll take care of the translation updates for you.

**Comment by Hartmnt** [comment]
I just compiled and tested the branch. The mp3 recording seems to work. VLC recognizes the resulting file as mp3
 
image

I would say it works as advertised

**Review by Hartmnt — APPROVED** [review]

**Comment by Krzmbrzl** [comment]
@Hartmnt Thanks for double-checking!

@Chinry thanks for implementing this feature!

**Pull request merged**